src/Workspaces/CoreTestUtilities/Remote/InProcRemoteHostClientProvider.cs

```diff
@@ -34,11 +38,17 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
 
         private sealed class WorkspaceManager : RemoteWorkspaceManager
         {
-            public WorkspaceManager(SolutionAssetCache assetStorage, Type[]? additionalRemoteParts)
+            public WorkspaceManager(SolutionAssetCache assetStorage, ConcurrentDictionary<Guid, TestGeneratorReference> sharedTestGeneratorReferences, Type[]? additionalRemoteParts)
                 : base(assetStorage)
             {
                 LazyWorkspace = new Lazy<RemoteWorkspace>(
-                    () => new RemoteWorkspace(FeaturesTestCompositions.RemoteHost.AddParts(additionalRemoteParts).GetHostServices(), WorkspaceKind.RemoteWorkspace));
+                    () =>
+                    {
+                        var hostServices = FeaturesTestCompositions.RemoteHost.AddParts(additionalRemoteParts).GetHostServices();
+
+                        ((IMefHostExportProvider)hostServices).GetExportedValue<TestSerializerService.Factory>().SharedTestGeneratorReferences = sharedTestGeneratorReferences;
```

Comment:
a quick comment here about how these references are coming from the MEF composition on the other side might be useful

Reply:
We share references in a slightly different way starting with #48044. Can we use the same approach for this?

Reply:
@sharwell: so this is hooking the serialization process in the same way as that PR. The goals though are different. Whereas that PR is simply trying to cache an thing that can be reused across multiple tests and the static is OK, the objects being shared here are unique per-test. Using a static would have been "easier" but would have been, if nothing else, a leak while tests are running.

Reply:
To @dibarbet's original comment: comment added.